src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/QueueItem.cs

```diff
@@ -56,59 +52,85 @@ internal class QueueItem<TRequest, TResponse, TRequestContext> : IQueueItem<TReq
         cancellationToken.Register(() => _completionSource.TrySetCanceled(cancellationToken));
 
         _logger = logger;
-        _request = request;
+        SerializedRequest = serializedRequest;
         LspServices = lspServices;
 
         MethodName = methodName;
-        Language = language;
 
         var telemetryService = lspServices.GetService<AbstractTelemetryService>();
 
         _requestTelemetryScope = telemetryService?.CreateRequestScope(methodName);
     }
 
-    public static (IQueueItem<TRequestContext>, Task<TResponse>) Create(
+    public static (IQueueItem<TRequestContext>, Task<object?>) Create(
         string methodName,
-        string language,
-        TRequest request,
+        object? serializedRequest,
         ILspServices lspServices,
         ILspLogger logger,
         CancellationToken cancellationToken)
     {
-        var queueItem = new QueueItem<TRequest, TResponse, TRequestContext>(
+        var queueItem = new QueueItem<TRequestContext>(
             methodName,
-            language,
-            request,
+            serializedRequest,
             lspServices,
             logger,
             cancellationToken);
 
         return (queueItem, queueItem._completionSource.Task);
     }
 
-    public async Task<TRequestContext> CreateRequestContextAsync(IMethodHandler handler, CancellationToken cancellationToken)
+    public async Task<TRequestContext> CreateRequestContextAsync<TRequest>(TRequest request, IMethodHandler handler, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
 
         _requestTelemetryScope?.RecordExecutionStart();
 
         var requestContextFactory = LspServices.GetRequiredService<AbstractRequestContextFactory<TRequestContext>>();
-        var context = await requestContextFactory.CreateRequestContextAsync(this, handler, _request, cancellationToken).ConfigureAwait(false);
+        var context = await requestContextFactory.CreateRequestContextAsync(this, handler, request, cancellationToken).ConfigureAwait(false);
         return context;
     }
 
+    public TRequest? TryDeserializeRequest<TRequest>(AbstractLanguageServer<TRequestContext> languageServer, RequestHandlerMetadata requestHandlerMetadata, bool isMutating)
```

Comment:
Previously, deserialization errors would never be observed by the queue, the request would just fail (but the server would continue functioning).

Now that we're deserializing in the queue, we log the errors and throw/catch based on what kind of request it is.